### ISSUE TITLE
Graphql Setup

### ISSUE THREAD

**Issue by jimjones26** [drejohnson/sapper-graphql-firebase#5]
Hey, thanks again for this template. I am a bit of a noob, and I am looking into setting up graphql in my sapper app using hasura. Would you still recommend this setup for graphql? I was looking at the svelte-apollo package but I like your approach better, I just don't know enough yet to know if there are any things to be aware of with this approach.

Thanks again for your time!

**Comment by drejohnson** [comment]
I would say if you need things like in-memory caching or subscriptions and peace of mind knowing it will work as intended, use Apollo. It's the gold standard right now. I just prefer a simpler and more lightweight approach. Apollo adds a lot of weight to your app. I love Svelte for how lightweight it is, so adding Apollo just seems wrong lol.

If all you need is queries and mutations, using the fetch api is a lot simpler imo. Plus you can kinda cache your graphql queries with service-workers if your willing to invest in a more DIY approach.

I started to build a lightweight graphql client for svelte but haven't had the time to really dig into it. Soon maybe.

**Comment by jimjones26** [comment]
Hey, thanks for the answer! You are right, every time I looked at adding apollo to my sapper app I just could not bring myself to do it. I also ended up just creating a simple composeFetch function last night / today that allows me to pass in queries and optional query vars and that seems to be working fine for now. I will dig through your code this weekend to see if I can understand what is going on with the caching. Thanks again for the response, I appreciate your time!
